app/main.py

```diff
@@ -1,2 +1,101 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.length = 0
+        self.capacity = 8
+        self.buckets = [None] * self.capacity
+
+    def _hashed_value_calculation(self, key):
+        hashed_value = hash(key)
+        return hashed_value % len(self.buckets)
+
+    def _resize(self):
+        self.length = 0
+        old_buckets = self.buckets.copy()
+        self.buckets = [None] * self.capacity * 2
+        self.capacity *= 2
+        for i in old_buckets:
+            if i is not None:
+                self.__setitem__(i[0], i[1])
+
+    def __setitem__(self, key, value):
+        index = self._hashed_value_calculation(key)
+        while self.buckets[index] is not None:
+            if key == self.buckets[index][0]:
```

Comment:
also, you should the hash

Reply:
![image](https://user-images.githubusercontent.com/72562232/193609314-a6ab2b09-37e9-463d-bfc7-95ba91520837.png)
different hash, but index the same

You should firstly check the hash

Reply:
I still don't understand what you mean by checking the hash?
I am calculating the hash and converting it to an index for the table in the `_hashed_value_calculation function`.
The hash itself is not stored anywhere in my dictionary.

Reply:
and&) you should store the hash in your dictionary)